src/overlays/actors/ovl_Dm_Hina/z_dm_hina.c

```diff
@@ -33,24 +33,156 @@ ActorInit Dm_Hina_InitVars = {
+        Actor_PlaySfx(&this->actor, NA_SE_OC_WHITE_OUT_INTO_KYOJIN);
+    }
+    Actor_PlaySfx(&this->actor, NA_SE_EV_MASK_RISING - SFX_FLAG);
+}
+
+void func_80A1F75C(DmHina* this, PlayState* play) {
+    s32 i;
+
+    switch (this->unk17E) {
+        case 0:
+            this->unk17F = 0;
+            this->unk14C = 0.0f;
+            break;
+
+        case 1:
+            Math_SmoothStepToF(&this->unk14C, 1.0f, 0.4f, 0.05f, 0.001f);
+            this->unk17F = this->unk14C * 255.0f;
+            this->unk150 = Math_SinS(play->state.frames * 0x1F40);
+            for (i = 0; i < 3; i++) {
+                play->envCtx.lightSettings.ambientColor[i] = play->envCtx.lightSettings.fogColor[i] =
+                    play->envCtx.lightSettings.diffuseColor1[i] = -255.0f * this->unk14C;
+            }
+            play->envCtx.lightSettings.fogNear = -500.0f * this->unk14C;
+            if (play->envCtx.lightSettings.fogNear < -300) {
+                play->roomCtx.curRoom.segment = NULL;
+            }
+            break;
+    }
+}
+
+void DmHina_Update(Actor* thisx, PlayState* play) {
+    DmHina* this = THIS;
+
+    this->actionFunc(this, play);
+    func_80A1F75C(this, play);
+}
+
+void func_80A1F9AC(DmHina* this, PlayState* play) {
+    GraphicsContext* gfxCtx = play->state.gfxCtx;
+
+    if (this->unk17E != 0) {
+        OPEN_DISPS(gfxCtx);
+
+        Gfx_SetupDL25_Xlu(play->state.gfxCtx);
+
+        gDPSetPrimColor(POLY_XLU_DISP++, 0, 0, 255, 255, (u8)(this->unk14C * 100.0f) + 155, this->unk17F);
+        gDPSetEnvColor(POLY_XLU_DISP++, 255, 255, (u8)(this->unk150 * 100.0f) + 50, 0);
+        gSPDisplayList(POLY_XLU_DISP++, gLightOrbMaterial1DL);
+
+        Matrix_Translate(this->actor.world.pos.x,
+                         this->actor.world.pos.y + (this->unk154 * this->unk15C) + (40.0f * this->unk15C),
+                         this->actor.world.pos.z, MTXMODE_NEW);
+        Matrix_ReplaceRotation(&play->billboardMtxF);
+        Matrix_Scale(this->unk14C * 20.0f, this->unk14C * 20.0f, this->unk14C * 20.0f, MTXMODE_APPLY);
+        Matrix_RotateZF(Rand_ZeroFloat(2 * M_PI), MTXMODE_APPLY);
+
+        gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        gSPDisplayList(POLY_XLU_DISP++, gLightOrbModelDL);
+
+        CLOSE_DISPS(gfxCtx);
+    }
+}
+
+void DmHina_Draw(Actor* thisx, PlayState* play) {
+    DmHina* this = THIS;
+
+    if (this->unk180 != 0) {
+        Matrix_Translate(this->actor.world.pos.x,
+                         this->actor.world.pos.y + (this->unk154 * this->unk15C) + (40.0f * this->unk15C),
+                         this->actor.world.pos.z, MTXMODE_NEW);
+        Matrix_RotateZYX(0, play->gameplayFrames * 0x3E8, 0, MTXMODE_APPLY);
+        Matrix_Scale((this->unk148 * (1.0f - this->unk14C) * this->unk15C),
+                     (this->unk148 * (1.0f - this->unk14C) * this->unk15C),
+                     (this->unk148 * (1.0f - this->unk14C) * this->unk15C), MTXMODE_APPLY);
```

Comment:
```suggestion
    DmHina* this = THIS;
    f32 scale_factor;

    if (this->unk180 != 0) {
        Matrix_Translate(this->actor.world.pos.x,
                         this->actor.world.pos.y + (this->unk154 * this->unk15C) + (40.0f * this->unk15C),
                         this->actor.world.pos.z, MTXMODE_NEW);
        Matrix_RotateZYX(0, play->gameplayFrames * 0x3E8, 0, MTXMODE_APPLY);
        scale_factor = this->unk148 * (1.0f - this->unk14C) * this->unk15C;
        Matrix_Scale(scale_factor, scale_factor, scale_factor, MTXMODE_APPLY);
```

Reply:
If this does work, I would argue just going with `scale` instead of `scale_factor`